### DISCUSSION TITLE
Obscure Nuxt 3 release roadmap

### DISCUSSION THREAD

**Discussion by WP-LKL** [nuxt/nuxt#9398] [General]
Firstly, thank you for this incredible framework, very powerful and comparatively much more operational -- good job!

However it is unclear when we can expect the first public release of Nuxt 3.
.. Asking about v. 3 usually prompts a response to view #5708, a thread that actually doesn't convey much about release progress both hinted within the slides and issue #9284 to be this month.

The first talks of Nuxt 3 began years ago shortly after Vue 3, and sparked [much false hope for a quick adaption of Nuxt 3](
https://github.com/nuxt/nuxt.js/issues/5708#issuecomment-668555312). And I am just asking you to consider that there are projects out there delaying front-end development in the anticipation of a Nuxt 3 release this month.

Forgive the impatience, I hope you can understand where this is coming from.

**Comment by codingcn** [comment]
Vue3 has been released for a year.
nuxtjs update too slow

**Reply by kissu** [reply]
Nuxt relies on a lot of things. Having them all upgraded and stable takes time. On top of that, the team added some few changes hence it will not be a simple bump from Vue 2 to 3.
A bit more patience should be appreciated IMO. I'd rather have the core team release something working than buggy and unstable as hell.

The transition to Vue3 is annoying in various ways but it should not be as much as a breaking change later on. So, try to wait a little bit more. 😄 

PS: Vue3 is not production-ready for a year AFAIK.

**Reply by reinoldus** [reply]
> PS: Vue3 is not production-ready for a year AFAIK.

Why do you think vue3 isn't production ready? The main components (router, store) are all vue3 compatible and vue3 is out of beta for quite a while

**Reply by kissu** [reply]
The main components are pretty stable but usually, you do need more on a real app and if some of those things are still not compatible, you can be hard blocked.

**Reply by nahojj** [reply]
Agree with you @kissu. I was just checking to see if we got any i18n ready and I found one package that seems to be working but after trying it out it failed pretty hard with VITE. In my opinion, it's still a very long way to go before we can say it's production-ready and stable... and to be honest, the past of pushing forward with these kinds of plugins/packages/modules doesn't seem to be here in the nearest future.

I wouldn't recommend anyone to use Nuxt3 for real production projects yet.

**Comment by calebeaires** [comment]
This [npm package](https://www.npmjs.com/package/nuxt3) refers to Nuxt 3. The [official slides from Nuxt teams](https://docs.google.com/presentation/d/1cWxCe9IZVRzM2W4OqvA4PtBr9_GLcF3n7VJEUoBLjg8/edit#slide=id.gbe7ff63d62_0_1443) say we going to see a beta release in Q2. Hope so.... I cant wait to test, even use it at production.


![CleanShot 2021-06-08 at 08 52 53](https://user-images.githubusercontent.com/794195/121180184-f2123200-c836-11eb-8ee4-76c61421bf5a.png)

**Reply by codingcn** [reply]
Q2 means June 30

**Reply by kissu** [reply]
@codingcn they still have 2 weeks left. 😉

**Reply by little3201** [reply]
any news?

**Reply by kissu** [reply]
@little3201 you remind me of the babies asking their parents when is the roadtrip over. Every 30s. 😊
Just wait for it while working on something else. 👍🏻

**Reply by tobiasdiez** [reply]
As parents you also quickly learn that explaining your kids where you are going and when you expect to be there is more effective than treating them with silence.

No one here is complaining that Nuxt 3 is not yet here (although everyone would probably be very happy if it would be), but it's the nontransparent process that people find irritating. Especially, given then open-source nature of Nuxt I'm somewhat surprised that Nuxt 3 developed behind closed curtains, instead of inviting people to contribute to the development (either in code or in feedback to say API design questions). I guess the core team has good reasons for this approach - but it would have been nice if they would at least be communicated.

Summary: Simply having an open roadmap with a list of todos that still needs to be done before Nuxt 3 is opened to the public, would silence all these excited kids that want to play with a new toy.

**Reply by kissu** [reply]
Traffic jam happens, so the parents don't always know when they do expect to arrive themselves. Better to say "soon" than to promise "30min" and having to apologize for being late and needing 2 extra hours.

Nuxt3 is currently developed as non open-source code because it is still in early stage. There will be a public bêta later on.

Vue3 was done the same way: Evan and his team worked on it. Then, when it was good enough, they released it to the public and then, everybody could give feedback about the last few things to polish. As of today, you can still give feedback [there](https://github.com/vuejs/rfcs) but it was not the case a year or two ago. There is no point into asking feedback when some things are related to the core implementation rather subjective choices.

Nuxt team is doing it's best currently and working hard (visited their HQ previous week). So don't think that they are smoking cigars and laughing about all the people being excited about the new version.
![image](https://user-images.githubusercontent.com/5133074/122767542-12b0a200-d2a3-11eb-8b5c-b67db93af295.png)

---
Disclaimer: I'm not part of the core team tho, but I still think that everybody is doing their best and are not making shady things. Pooya Parsa did his best about giving an estimated data during his latest presentation (linked above). 
For a basic reference, I guess that you can follow this [**unofficial**](https://unofficial-nuxt3-countdown.netlify.app/) release plan and also follow [Sébastien Chopin](https://twitter.com/Atinux).

**Reply by RedKage** [reply]
If that's the case, then why even make a roadmap? Why not just say _"when it's done"_?
Here the issue is the very vague roadmap which creates expectations.

Why tell your kids in the car the remaining time when you became lost on the wrong freeway and have no idea when you can take the next exit? At that point it's better to say to forget the estimates.

I am stuck as well on a project where we need Vue3 on Nuxt, mostly for performance reasons.
And, no, I cannot "work on something else".
I had to decide if we could wait Nuxt3 or make a workaround which would be costly in time.
With a clear roadmap our decision would have been easier to take:
- Wait if the release was last month / early this month
- No wait if later

In the end my intuition told me that Nuxt3 won't be out anytime soon and informed my team Nuxt3 will most likely be for Q2 next year seeing how poor the communication was.

Thanks for the unofficial release plan, this looks much more realistic!

**Reply by kissu** [reply]
@RedKage at first, this was the actual answer: "when it's done". The team started talking about a roadmap probably because people were spamming for it. I get the _why_ for both parties tho.

Telling your kids "soon" is still the best thing since it's vague and you never know if one of your wheels randomly dies. If people will ask the devs to respect the timelines, I guess that we will have a rushed release with a lot of bugs (heard of `Cybperpubk 2077`? yep it was shit and hit hard their studio because of MaRkEtInG reasons you know 🤑).

Gaming and tech is pretty much the same, you need to have a balance between:
- releasing something at the due data (otherwise people will be angry for you being late)
- not releasing something properly tested and working well (otherwise people will be angry because the code is meh)
- having a life and not working 23h/day (people can still be angry because you do have a burnout and stop releasing stuff + family issues)

So far, the best decisions lately were to postpone any tech product to a later deadline to ship a refined and cool product. 👍🏼 

---
There are some performance benefits with Vue3, meanwhile this would matter if your project is super huge and that you've already checked all the other possible sources from which it may come from. Aka, it being Vue2 is probably not a blocker in itself: try to look for other aspects of your app having performance issues.

When Nuxt3 will get released, you'll probably have some other modules that are not totally battle-tested and working, for a production release at least. So if you are bleeding edge on your versions of Vue3, you should rather implement some SSR in a different way. If you're looking for performance only, look somewhere else IMO.

It's our job as developers to find the right balance and take decisions. Maybe your app can be totally snappy with Vue2, just need to find out why. Maybe the whole Vue ecosystem is crap and you should have something in Svelte or in Web Components. Maybe you could hire an expert for some homemade solution to your performance issues.

Relying on a team to release a huge major version is not a solution in itself if you're using this as your daily production product. You could start to write your code the Vue3-way today, for it to be composition API's compliant tomorrow and stay in this middle ground for some time. Or wait for the stable release of Nuxt3 but even running this in public-beta is a bad idea and it'll take some time to get refined.

**Reply by WP-LKL** [reply]
@kissu Kindly consider that everybody here understands the obvious caveats; that deadlines are not bulletproof, they [the NuxtJs team] have no obligations and their own lives, that things can go wrong etc. 

Again, we are merely humbly submitting an inquiry to the hardworking people of NuxtJs to comment on release progress supposedly to commence within 10 days (25 days by the time of posting), which scarcely can be considered unfair given the history of wildly inaccurate deadlines as alluded to in the OP.

A simple comment would serve as a great relief for those - such as the people above - who are making decisions about how they should develop skills and/or products in a framework developed behind closed-doors and with limited or misleading (hazardous) communication as to its state.

**Reply by kissu** [reply]
@WP-LKL I see that you get the obvious, it's not the case for everybody. Thanks for your patience Lars!

I do agree on the fact that some feedback needs to be brought up to improve the framework and that some informations may go down to let the people know how things go. It's lacking on this part.

I would do my best to share some of this feedback to the team when I'll get the opportunity. They are quite interested in any feedback (documentation, issues etc..) and are just missing a man in the middle for this. I think that `Nuxt.js discussions` will get more love in the future.

I'll do my best to improve the relay between both parties in the next weeks/months.

**Reply by calebeaires** [reply]
👀

**Reply by RedKage** [reply]
@kissu 
Lots of assumptions.
My perf issues were indeed related to Vue2 only, it's about tearing down reactivity dependencies.
And yes my app is huge, so yes it does matter.
To bad it's me "the expert" about performance.

Have a look, this is a known performance issue
- https://forum.vuejs.org/t/strange-performance-reading-change-one-line-and-performance-says-kabum/44567/5
- https://forum.vuejs.org/t/chromium-based-browser-and-reactivity-before-destroying-lot-of-components/116165
- https://github.com/vuejs/vue/pull/8157
- https://github.com/vuejs/vue/issues/10435
- https://github.com/vuejs/vue/pull/8944

From what I gathered Vue3 will be using a proxy system for its reactivity dependencies which should make all of this faster.
In the end we managed without Vue3. But that would have been nice to see if out of the box these perfs issues were gone with Vue3, also giving us hints about the real issue on our end. Instead of debugging for about a week and realizing that, yes, that's was a Vue2 reactivity "issue".

We stabilized our app in the meantime, doing lots of stuff including removing features. Kinda.
And did some refactoring which degraded readability because some reactivity stuff (like computed) had to be removed.
I know you are going to say "you don't want to have huge lists anyway, bad design" so no, it wasn't a huge list, just about 20 items. Each of them having about 10 computed stuff related to our store. Yes we need these in the store. Tearing down these 20 items would freeze the browser for 30 seconds depending on browser, RAM and other computer specs. Crazy.

Anyway I digress, I am eager to see even more perf gain with Nuxt3 hopefully!

Thanks for giving us more info about the roadmap, I appreciate it.
Cheers

**Reply by kissu** [reply]
> "you don't want to have huge lists anyway, bad design"

Lot of assumptions here too. Nah, just kidding! 😛
Sorry for mines btw.

I get what you're saying and hope that the performance issues will be solved with Vue3. 😄 
As for the communication from the Nuxt team, we will try to do something nice especially since Stackoverflow just announced Collectives: https://stackoverflow.com/collectives-on-stack-overflow/tour

Have a great day! ❤️

**Reply by lampewebdev** [reply]
It would be just nice to know what is happening.

We looked at the roadmap and planed accordingly to that roadmap. Of course not 100 % on that timing but at least to see a early early beta around this time now. Just something we could try to integrate and see how it turns out.

As of today since we don't know what the roadmap will be we are forced to change our plans of using nuxtjs.

And we will probably go with vue3 SSR directly or vite.

@kissu this has nothing to do with crying its just business and reliance.
If we see that we can not rely on something, it is just a smart move to look for alternatives.
I don't get the comparison to game development since this is a different field.
Also there are countless examples of games just stuck in development hell.
Making a game good after X months is usually a sign that it will still suck.
Also we are talking here about an beta release not a finale product.

> So far, the best decisions lately were to postpone any tech product to a later deadline to ship a refined and cool product. 👍🏼

Not really its just that you dot hear about the 1000 of failures.

**Reply by kissu** [reply]
@lampewebdev you can already use [composition API in Nuxt](https://composition-api.nuxtjs.org/getting-started/introduction) as of today and then, make a smooth transition once Nuxt3 is released so far.

Also, [Quasar v2](https://github.com/quasarframework/quasar/discussions/9734) just released few days ago, this is another viable solution if you need Vue3 ASAP. Or the alternatives that you've mentioned yeah!

I get your point on deadline's planning, but sh*it happens. 😄 
Also, sorry if my comment was mean, didn't meant to be agressive.
Was just a long day I guess (at 2pm yeah 😂).

The game industry comparison was essentially to point out that the last 18 months were pretty hardcore human-wise and that everything can happen when you're dealing with software, life will not await that you've done your work to strike you in the ribs with some bad news.

Sometimes we do hear about fails but the biggest ones are not silent. Of course there will be others, smaller ones.
What I meant is that your business will not crash if you have your code still written with the Options API, and hundreds of families will not be left at the door because of it. (**assumption here**, hope it's not the case 😱)


So yeah, we are all hyped devs trying to get the latest tasty tool to play with. Let's just wait for Nuxt team to finish their work. They will maybe send a pack of stickers for all the eager developers who spoke up here! 🎊

PS: looking at Vue3 as it is right now, I guess that Nuxt3 will not be too buggy once released, even as a beta.

**Reply by lampewebdev** [reply]
@kissu
Its not just about the composition Api.
Vue3 has more things then that that we need.

And we don't expect that nuxt3 will be bug free.

Our problem as of today is that we have a tested setup of webpack5 and we have development teams that are waiting for nuxt3 to start testing stuff. Like our own component library which should work with vue3 and nuxt3.

In our blocker list we had storybook, vue-router and vuex. They all now work with vue3 even if they are only in rc. We still can test them.

 We are also not expecting a long blog post or a long video. Just some information. Like an updated roadmap or just a tweet: Nuxt3 will be late expect it in q4.

Done. Yes there will be angry people but you have them also now.
So yeah I don't fully understand the strategy here of the nuxtjs team.

**Reply by kissu** [reply]
@lampewebdev if you have done a lot of work yourself with Webpack5, Vuex, router and so on, why not simply make your own "Nuxt" at this point?
You could keep it while the official Nuxt3 is released. Nuxt is essentially a configuration on top of Vue but if your team is moving fast and if this is the only and last blocking part of your team, make your own setup.
At the end, Nuxt is a great SSR standard that people agree with, if this standard is blocking your team, move around it. Some libraries are monkey-patched like this to allow for more flexibility or when PRs take ages to be merged.

If done properly, once Nuxt3 is released and stable, you could strip your own solution by the official package.
Even better: if you do have some cool ideas, you could bring them to this project and everybody could move forward (faster) ! ✨ 💪🏼 🎊

**Reply by nelsonlarocca** [reply]
IMO Nuxt Team is absolutely professional so I trust they will release the product when really stable.

On the other side, I would like to remember to all people that this framework is super-organized, and it's been fantastically growing  up, and that is absolutely FREE, so there's no place to complain.

Consider also that since the sixties when, the software engineering was born creating new management methods and techniques to improve efficiency and even that, projects take more time than predicted, no matter how many people working on it.

In the meanwhile, let's enjoy all the Nuxt power we have.
Cheers,
Nelson

**Reply by kissu** [reply]
There are currently [76 people supporting Nuxt](https://github.com/sponsors/nuxt) through Github Sponsors, with the first tier being 10$.
@nelsonlarocca is right on the fact that you can sponsor the project if you're generating money thanks to it, an okay payback for the hard work of the team there.
As Wes Bos said in one of it's latest podcasts, if you're donating to open source, you're helping yourself by giving more resources to a company that will be able to deliver better software. Software that you'll use back on your daily job at some point (virtuous circle).

If you give something, the team will be able to hire somebody else to help them code-wise, communication-wise or improving the documentation etc... If you don't feel like sponsoring, there is no pressure and it's totally fine. 👍🏼 
Meanwhile, try to be kind and not put the pressure on the team neither. 🕊️

**Comment by tobiasdiez** [comment]
As secretly announced on the preview of the new nuxt page https://preview.nuxtjs.org/3 (nuxt/nuxt, see https://github.com/nuxt/nuxt.js/discussions/9535), the release plan seems to be as follows:
- July 2021: Private beta for Vue core team and Nuxt modules authors.
- August 2021: Public beta.
- Q4 2021: Stable version

**Reply by calebeaires** [reply]
I'm not a member of the dev team neither a plugin maker... but I wish I could...

**Reply by kissu** [reply]
@calebeaires it's actually publicly available. 👌🏻

**Reply by calebeaires** [reply]
@kissu help me get there

**Reply by kissu** [reply]
@calebeaires you can the repo here: https://github.com/nuxt/nuxtjs.org/tree/preview
This is not the actual Nuxt app, but you can probably still inspect and look for some things before the others. 😄

**Comment by cprcrack** [comment]
I don't have an issue with the Nuxt 3 release still taking several months. Honestly the community should be so grateful that Nuxt even exists, being patient is the least we can all do.

However, it's true that the lack of a public roadmap makes it feel like a less professional and dependable project. Take for example Vuetify 3, it's also taking them more than a year to release, but there is a clear roadmap which just gives confidence on the project: https://vuetifyjs.com/en/introduction/roadmap/

Having some public information out there with an official roadmap, even if it's just a couple of paragraphs, would render discussions such as this one unnecessary, which I think is good for Nuxt overall.

**Reply by calebeaires** [reply]
Good suggestion!

**Comment by hkmsadek** [comment]
No update on nuxt 3? Any date we can expect?

**Reply by reinoldus** [reply]
Sebastien Chopin will speak at the nuxt nations conference: https://nuxtnation.com/

Given that the release is scheduled for September it could be somewhat likely that this will be used as a kick-off event.

**Reply by Saganic** [reply]
I'm just about to begin a re-write on a medium sized production Vue SPA in Nuxt with the composition API. Reckon it's worth waiting?

**Reply by nandi95** [reply]
Probably not.
I would use nuxt 2 with composition api and if you lucky to be using a component library that has a vue 3 version and doesn't differ a lot then upgrading won't be a lot of pain.

**Reply by Miguelklappes** [reply]
> 
> 
> I'm just about to begin a re-write on a medium sized production Vue SPA in Nuxt with the composition API. Reckon it's worth waiting?

Might be delayed to October, then to November, and so on

**Comment by aspearman** [comment]
**Q4 2021** is when it's slated for stable release. See https://preview.nuxtjs.org/v3 with user/pass `nuxt`/`nuxt` (#9535)

<img width="1162" alt="Screen Shot 2021-08-27 at 11 44 24 PM" src="https://user-images.githubusercontent.com/27811981/131122359-59e67cc4-0c80-4260-bb3d-763184096bfb.png">

**Reply by manigandham** [reply]
Unfortunately these timelines have changed so often without any other communication that they've become meaningless. At this point it'll be released when it's released.

**Reply by aspearman** [reply]
I included the [link](https://preview.nuxtjs.org/v3) so that you can keep up to date with any timeline changes. The point is there is a public timeline now 🚀

**Reply by manigandham** [reply]
The site is still in preview (not fully public) and it's already been shared before [in this discussion (above)](https://github.com/nuxt/nuxt.js/discussions/9398#discussioncomment-988146) and elsewhere: https://www.reddit.com/r/Nuxt/comments/oye7i2/nuxt_3_megathread/

To be clear, the problem isn't finding a timeline but that nothing happens as each deadline passes by, and instead the timeline gets silently changed.

**Reply by kissu** [reply]
Nuxtnation is coming soon, check this for some announcement. 😉

**Reply by Miguelklappes** [reply]
> 
> 
> Nuxtnation is coming soon, check this for some announcement. 😉

Pretty sure Nuxt still does not work on Windows, so they will not be releasing a public beta anytime soon.

**Reply by kissu** [reply]
> Pretty sure Nuxt still does not work on Windows, so they will not be releasing a public beta anytime soon.

I'm not sure that Windows is a top priority as of right now. The beta will probably not support it.

Do you use it on Windows as of right now? What kind of issue do you have there?

**Reply by Miguelklappes** [reply]
Well, it should be a priority. The issue is the path used in Windows, a simple function would be the fix, I don’t see how could they launch a public beta without supporting Windows.

I could still use WSL2 or a dual-boot in MacOS/Linux, but that’s just probably me and a few other people.

**Reply by little3201** [reply]
要不是因为vue技术栈，早都放弃等待nuxt3了

**Reply by codingcn** [reply]
As an open source project, nuxt3's release plan is really bad

**Reply by carbotaniuman** [reply]
I'm close to jumping ship to Vite's SSR because of how opaque the entire Nuxt 3 process has been. This isn't a single maintainer working on the project for fun, and yet the experience waiting for Nuxt 3 feels more like waiting for proprietary software to update than collaborating with the community.

**Reply by zernonia** [reply]
> > Pretty sure Nuxt still does not work on Windows, so they will not be releasing a public beta anytime soon.
> 
> I'm not sure that Windows is a top priority as of right now. The beta will probably not support it.
> 
> Do you use it on Windows as of right now? What kind of issue do you have there?

I'm so sad that I can't play with nuxt3 at the moment. I clone their starter v3  template, run `yarn install` then `yarn dev`, this is (probably 1 of the many) issues. 
I hope they will fix the Windows issue during public beta.

```sh
√ Nitro built in 219 ms                                                                16:35:38
√ Client Compiled successfully in 146.04ms                                             16:35:38
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM 
loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:782:11)
    at Loader.resolve (internal/modules/esm/loader.js:86:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:230:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:56:40)
    at link (internal/modules/esm/module_job.js:55:36) {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}
√ Server Compiled successfully in 60.94ms   
```

**Reply by Anoesj** [reply]
I can't find the Timeline section displayed in the image above anywhere on the new Nuxt website. Was it removed and if yes, is there a public roadmap somewhere else to find? Thanks!

**Comment by kevhof** [comment]
I'm sure it will be published in a few days, right before the [Nuxt Nation Conference](https://nuxtnation.com) on the 15th.

**Reply by pi0** [reply]
Nuxt 3 + bridge will finally become public sometime between Nuxt Nation and October. Expect more updates and spoilers sooner! Thanks for your patience and support 💚

**Reply by kissu** [reply]
We LOVE to be spoiled. ♥

**Comment by jonybekov** [comment]
Ah can't wait

**Comment by atinux** [comment]
I am sorry for the lack of communication regarding Nuxt 3 roadmap, it was a bumpy road but we finally tabled for a release date (public beta) that I will announce tomorrow at NuxtNation.

Honestly it was a though decision to start privately but we wanted to avoid the mistakes we did in the past with Nuxt 2. So Nuxt 3 can be versatile enough for the future to easily customize it. As well as making sure the transition between Nuxt 2 and Nuxt 3 can be progressive and smooth (we really don't want to create an Angular 2 feeling). Vite, Webpack 5 and Vue 3 does not help on this equation for sure 😅

Yes, the roadmap is hard to predict but I really apologize about the lack of public communication regarding Nuxt 3 development and I will do my best that this situation does not happen anymore in the future.

The team behind it worked really hard and we are all both excited and nervous to release it publicly (we know that there is a lot of expectations).

You can also take a look at @unjs organisation, a lot of repositories comes from the refactoring of Nuxt 3 into maintainable packages with a single purpose. If you want to help and contribute, there are already many issues that would love a pull request attached to it.

Also, the new website will help about adding updates frequently to keep everyone updated into a single place (no need to find some issues or discussion on GitHub or Twitter).

Again, thank yo for your trust in us and Nuxt 💚

**Reply by codingcn** [reply]
Another month later? October 12, 2021, 10 PM GMT+8

**Reply by EduardoMateos** [reply]
So I can migrate from Nuxt 2 to Nuxt 3 in a simple way? thanks!

**Reply by atinux** [reply]
We will do our best for making the migration smooth

**Reply by ctwhome** [reply]
thank you for the enormous effort!

**Reply by Patrick-Remy** [reply]
Are/will be slides or a video of the Nuxtnation session "Nuxt 3 Update" available?

**Reply by GillyRabutTsurwa** [reply]
@Patrick-Remy, Sebastian just talked about in the beginning of the NuxtNation conference. He also said there will be more to come in a month, when Nuxt 3 public beta is predicted to be released

https://nuxtjs.org/v3/

**Reply by kissu** [reply]
> Are/will be slides or a video of the Nuxtnation session "Nuxt 3 Update" available?

All videos will be available on YouTube. 😉

**Reply by WP-LKL** [reply]
@Atinux Thank you, that's very much appreciated!

**Comment by killix** [comment]
I feel really weird about the fact we can't test and maybe contribute to the v3.  `Private beta` what does this means for an opensource project? Do you plan to change the Licence, this is only the reason I can find to explain this...

**Reply by kissu** [reply]
It is probably because the team has some specific parts of the core that they have a vision about and want to just be focused on this one while not losing time with community feedback at this level of the project. Not all the parts require community feedback from the bigger crowds. Those who already contributed once to the project can have access to it. As usually, if you want to have a voice in the process, the best way is to start contributing. 👍🏻

If you are afraid of it changing licence, don't be because it will be the same one (MIT). Also, please remember that Vue3 was in private beta by Evan You at first too, for the same reasons. 👌🏻

**Reply by kennylajara** [reply]
You can now play with the Public Beta version: https://v3.nuxtjs.org/getting-started/introduction

**Comment by antonreshetov** [comment]
According to this [tweet]( https://twitter.com/vuejs/status/1484160249686749191?s=21) Feb 7, 2022 Vue 3 will become the default version. Can we expect Nuxt 3 to follow this date or is there any indication of a specific release date for the stable version?

**Reply by reinoldus** [reply]
Most likely not, since nuxt3 is not stable yet

**Reply by Anoesj** [reply]
Yes, a little more information about the state of Nuxt 3 (and progress on modules) would be nice. We know this is OSS, so it’s not our place to make demands, but from a outsider’s perspective, I’d prefer improved provision of information over fast releases or more features.

**Reply by antonreshetov** [reply]
I'm probably going to be hated, but I still want to say something. OSS on which donate, and this already imposes an obligation...

**Reply by kissu** [reply]
@antonreshetov I did not get this last part.

**Reply by ctwhome** [reply]
> Yes, a little more information about the state of Nuxt 3 (and progress on modules) would be nice. We know this is OSS, so it’s not our place to make demands, but from an outsider’s perspective, I’d prefer improved provision of information over fast releases or more features.

If it is a matter of money, maybe we should make an en effort, as a community, to support economically the project to have full developers working on it. But to make the project attractive, the financials as well its progress should be more transparent. 

There is also a lot of work on PR and getting onboard companies as sponsors.

**Reply by nicod-pc** [reply]
> Most likely not, since nuxt3 is not stable yet

Hi @reinoldus, what makes it unstable? Is there a list of known bugs that have to be fixed before declaring it stable?
There can still be unknown bugs in stable versions, so in most projects it's just a random point in time where projects decide it's now tested enough to make a stable release. Do you know more details about the nuxt approach to this?

**Reply by reinoldus** [reply]
@iwt-nduesing 
It says so on the website:
![image](https://user-images.githubusercontent.com/2091290/151532731-3ccb2921-59f3-497c-8a17-0e90359bb173.png)

Also it's not feature complete yet, hybrid and devtools are not ready yet.

Further if you read on twitter, reddit etc.. about experience reports you'll find that lot's of people hit issues.

Other than that I do not know anything else about the state of nuxt3. I am as affect by the "wall of silence" as all of you.

**Reply by matthieu-rolland** [reply]
Well, it's really hard to put release dates on open-source projects...

But given the issues and features not yet implemented,  I wouldn't expect a stable release before a few months

**Reply by danielroe** [reply]
Release candidate is slated for end of Q1 of this year, with final release out in Q4.

**Reply by antonreshetov** [reply]
> @antonreshetov I did not get this last part.

What I mean is that people donate to the project. Maintainers get those donations.Some maintainers are working full-time at OSS because of these donations. And since that's the case, I think people want to know at least the roadmap and release dates.

Of course donations are voluntary, but I doubt that maintainers will have the enthusiasm to keep working without donations.

**Comment by vaiil** [comment]
Hello, are there any dates of feature implementation, e.g static site generation in nuxt 3. Where can I follow the progress?

**Reply by danielroe** [reply]
Static site generation can be followed at https://github.com/nuxt/framework/discussions/560. RFCs and issues in general can be followed in that repository.

**Comment by bonomite** [comment]
I will be starting a NEW build soon... and want to use Nuxt3, how big of a risk would it be to use the current version?

**Reply by nahojj** [reply]
Well... I would never use a beta-version in production if the project is for a client and not a private one. I've been following and stayed tuned about when Nuxt will be ready and it seems like it will take some time to get all stuff in place.

**Reply by ctwhome** [reply]
The nuxt team is making a clear effort to share more info. See the tweets from @Atinux! 💚

**Reply by reinoldus** [reply]
That is definitely progress but I can't help it and gotta say it feels like torturing us with tweets.

Why can't there be a clear roadmap? A short blog article each month that summarizes what's going on?

**Reply by ctwhome** [reply]
I know that roadmaps are very difficult, and they can add a lot of unnecessary stress to the team. 
I do agree that a monthly small newsletter with info on what is going on, and planned features would be awesome. 
And of course, if the team is limited in capacity and need more people or money, then as a community we should support it

**Reply by reinoldus** [reply]
A roadmap can go either way, it's also not so easy to maintain. 

Anyway: Right now you have to follow all of the core developers on twitter and subscribe to every discussion/issue that is related to the state of nuxt. That is not ideal and very frustrating.

Also: I find it a bit odd that announcements like RC planned for March are only made on twitter...

**Reply by bonomite** [reply]
oh well... I guess if I start with Bridge and can transition to 3 at some point soon I hope

**Reply by ctwhome** [reply]
Every time I see this timeline I always think that something like that would be awesome in the nuxt 3 landing page as well. They are also open source and it does a good job keeping the expectations aligned https://tauri.studio/

**Comment by kissu** [comment]
Alright here is a public roadmap guys, enjoy ! 💚
https://twitter.com/nuxt_js/status/1498691921941872647?t=uKfHulfgjrh0UtojoKJsSg&s=19

Or the direct link: https://v3.nuxtjs.org/community/roadmap/

**Reply by Suniron** [reply]
Great news!!! Thanks 😄👍
